### ISSUE TITLE
Ploting att_ws maps would take a considerable amount of time

### ISSUE THREAD

**Issue by GuanghuiSong** [espnet/espnet#5194]
**Describe the issue**
When I used the Alimeeting dataset for speaker diarization tasks (using librimix/diar1), I found that ploting att_ws maps during the training phase would take a considerable amount of time (about 1 hour). May I ask what are the reasons for this? Where can I add code to remove ploting the att_ws?

**THis is my config.yaml**
```javascript

``` config: conf/train_diar_eda_5.yaml
print_config: false
log_level: INFO
dry_run: false
iterator_type: chunk
output_dir: exp/diar_train_diar_eda_5_raw
ngpu: 1
seed: 0
num_workers: 1
num_att_plot: 3
dist_backend: nccl
dist_init_method: env://
dist_world_size: 4
dist_rank: 0
local_rank: 0
dist_master_addr: localhost
dist_master_port: 53598
dist_launcher: null
multiprocessing_distributed: true
unused_parameters: false
sharded_ddp: false
cudnn_enabled: true
cudnn_benchmark: false
cudnn_deterministic: true
collect_stats: false
write_collected_feats: false
max_epoch: 250
patience: null
val_scheduler_criterion:
- valid
- loss
early_stopping_criterion:
- valid
- loss
- min
best_model_criterion:
-   - valid
    - acc
    - max
keep_nbest_models: 10
nbest_averaging_interval: 0
grad_clip: 5
grad_clip_type: 2.0
grad_noise: false
accum_grad: 1
no_forward_run: false
resume: true
train_dtype: float32
use_amp: false
log_interval: null
use_matplotlib: true
use_tensorboard: true
create_graph_in_tensorboard: false
use_wandb: false
wandb_project: null
wandb_id: null
wandb_entity: null
wandb_name: null
wandb_model_log_interval: -1
detect_anomaly: false
pretrain_path: null
init_param: []
ignore_init_mismatch: false
freeze_param: []
num_iters_per_epoch: null
batch_size: 16
valid_batch_size: null
batch_bins: 1000000
valid_batch_bins: null
train_shape_file:
- exp/diar_stats_16k/train/speech_shape
- exp/diar_stats_16k/train/spk_labels_shape
valid_shape_file:
- exp/diar_stats_16k/valid/speech_shape
- exp/diar_stats_16k/valid/spk_labels_shape
batch_type: folded
valid_batch_type: null
fold_length:
- 80000
- 800
sort_in_batch: descending
sort_batch: descending
multiple_iterator: false
chunk_length: 200000
chunk_shift_ratio: 0.5
num_cache_chunks: 64
chunk_excluded_key_prefixes: []
train_data_path_and_name_and_type:
-   - dump/raw/simu/data/train_4spk/wav.scp
    - speech
    - sound
-   - dump/raw/simu/data/train_4spk/espnet_rttm
    - spk_labels
    - rttm
valid_data_path_and_name_and_type:
-   - dump/raw/simu/data/dev_4spk/wav.scp
    - speech
    - sound
-   - dump/raw/simu/data/dev_4spk/espnet_rttm
    - spk_labels
    - rttm
allow_variable_data_keys: false
max_cache_size: 0.0
max_cache_fd: 32
valid_max_cache_size: null
exclude_weight_decay: false
exclude_weight_decay_conf: {}
optim: adam
optim_conf:
    lr: 0.002
scheduler: warmuplr
scheduler_conf:
    warmup_steps: 30000
num_spk: 4
init: xavier_uniform
input_size: null
model_conf:
    diar_weight: 1.0
    attractor_weight: 1.0
use_preprocessor: true
frontend: default
frontend_conf:
    fs: 16k
    hop_length: 128
specaug: specaug
specaug_conf:
    apply_time_warp: false
    apply_freq_mask: true
    freq_mask_width_range:
    - 0
    - 30
    num_freq_mask: 2
    apply_time_mask: true
    time_mask_width_range:
    - 0
    - 40
    num_time_mask: 2
normalize: global_mvn
normalize_conf:
    stats_file: exp/diar_stats_16k/train/feats_stats.npz
encoder: transformer
encoder_conf:
    input_layer: linear
    num_blocks: 4
    linear_units: 512
    dropout_rate: 0.1
    output_size: 256
    attention_heads: 4
    attention_dropout_rate: 0.1
decoder: linear
decoder_conf: {}
label_aggregator: label_aggregator
label_aggregator_conf: {}
attractor: rnn
attractor_conf:
    unit: 256
    layer: 1
    dropout: 0.1
    attractor_grad: true
required:
- output_dir
version: '202304'
distributed: true


encoder: transformer
encoder_conf:
    input_layer: "linear"
    num_blocks: 4
    linear_units: 512
    dropout_rate: 0.1
    output_size: 256  # dimension of attention
    attention_heads: 4
    attention_dropout_rate: 0.1

attractor: rnn
attractor_conf:
    unit: 256 # same as encoder output size
    layer: 1
    dropout: 0.1
    attractor_grad: True


optim: adam
grad_clip: 5
max_epoch: 250
optim_conf:
    lr: 0.002
scheduler: warmuplr
scheduler_conf:
    warmup_steps: 30000


best_model_criterion:
-  - valid
   - acc
   - max
keep_nbest_models: 10


iterator_type: chunk
chunk_length: 200000
num_cache_chunks: 64


batch_type: folded
batch_size: 16

init: xavier_uniform

specaug: specaug
specaug_conf:
    apply_time_warp: false
    apply_freq_mask: true
    freq_mask_width_range:
    - 0
    - 30
    num_freq_mask: 2
    apply_time_mask: true
    time_mask_width_range:
    - 0
    - 40
    num_time_mask: 2

**Comment by 791428954** [comment]
I'm sorry for copying the configuration file so crudely

**Comment by sw005320** [comment]
Can you try "num_att_plot: 0"?
